### DISCUSSION TITLE
Duplicate page structure for translations

### DISCUSSION THREAD

**Discussion by Jimmylet** [directus/directus#11804] [Archive]
Hello,

I have a builder that allows me to build my pages with different elements.  

In many CMS, when I decide to translate my document, my structure and the content of the fields are copied to the requested language.  The big advantage is that the user does not have to redo his entire layout: he simply has to edit the text.

I haven't found how to implement this with directus.  Is it possible ?

Concretely, if I have a page with 3 constructor elements added in a specific order (image, text, image), I would like the translation to find the same thing (image, text, image) without having to add them manually.

This is very useful when you have a lot of data.

Thank you for your help

**Comment by xu4wang** [comment]
It can be done by https://docs.directus.io/configuration/relationships/#translations-o2m

- Create a translation field in the data model (for your case, it will be Pages)
<img width="1309" alt="image" src="https://user-images.githubusercontent.com/311397/155255561-7899be2f-9436-423c-9696-ec1b5a870911.png">

- Add all the text and image  fields need to have multi language to the junction collection
<img width="1251" alt="image" src="https://user-images.githubusercontent.com/311397/155255592-3da2a621-a2e8-4c09-8583-11c2bdbd3ae8.png">

- Customize the data in the Languages collection, leave only the languages you need.

<img width="1267" alt="image" src="https://user-images.githubusercontent.com/311397/155255627-1985a963-dca1-4b26-8e06-92727c594348.png">

- It's done.

<img width="1251" alt="image" src="https://user-images.githubusercontent.com/311397/155255508-0001dad2-3d62-4330-95c4-356dfefc903d.png">

**Comment by Jimmylet** [comment]
Hello @xu4wang 

Thanks for your help !
I already knew this technique.

My problem is different:

If I create a flexible layout on demand, the fields created in French (in the edit view of the page) are not duplicated towards the other languages.

The user must redo all its flexible layout for each language. If it is a long enough page and there are 2-3-4 languages, it takes considerable time.

The flexible layout is created thanks to M2A fields

**Reply by xu4wang** [reply]
Sorry. how is the user experience for "flexible layout"?  would you please share a screen capture?

**Comment by rijkvanzanten** [comment]
If you setup translations on the "page" level, you would indeed recreate the page for every language. However, you can also add the translations on the "section" you're relationally pulling in , at which point you would translate each section's content individually 🙂
